desc/geometry/core.py

```diff
@@ -158,6 +158,77 @@ def __repr__(self):
             + " (name={})".format(self.name)
         )
 
+    def to_FourierXYZCurve(self, N=None, grid=None, phis=None, name=""):
+        """Convert Curve to FourierXYZCurve representation.
+
+        Parameters
+        ----------
+        N : int
+            Fourier resolution of the new X,Y,Z representation.
+        grid : Grid, int or None
+            Grid used to evaluate curve coordinates on to fit with FourierXYZCurve.
+            If an integer, uses that many equally spaced points.
+        phis : ndarray
+            angle to use for the fitting. if None, defaults to
+            normalized arclength
+        name : str
+            name for this curve
+
+        Returns
+        -------
+        curve : FourierXYZCurve
+            New representation of the curve parameterized by Fourier series for X,Y,Z.
+
+        """
+        from .curve import FourierXYZCurve
+
+        coords = self.compute("x", grid=grid, basis="xyz")["x"]
+        return FourierXYZCurve.from_values(
+            coords, N=N, phis=phis, basis="xyz", name=name
+        )
+
+    def to_SplineXYZCurve(self, knots=None, grid=None, method="cubic", name=""):
+        """Convert Curve to SplineXYZCurve.
+
+        Parameters
+        ----------
+        knots : ndarray
+            arbitrary theta values to use for spline knots,
+            should be an 1D ndarray of same length as the input.
+            (input length in this case is determined by grid argument, since
+            the input coordinates come from
+            Curve.compute("x",grid=grid))
+            If None, defaults to using an equal-arclength angle as the knots
+            If supplied, will be rescaled to lie in [0,2pi]
+        grid : Grid, int or None
+            Grid used to evaluate curve coordinates on to fit with SplineXYZCurve.
+            If an integer, uses that many equally spaced points.
+        method : str
+            method of interpolation
+            - `'nearest'`: nearest neighbor interpolation
+            - `'linear'`: linear interpolation
+            - `'cubic'`: C1 cubic splines (aka local splines)
+            - `'cubic2'`: C2 cubic splines (aka natural splines)
+            - `'catmull-rom'`: C1 cubic centripetal "tension" splines
+        name : str
+            name for this curve
+
+        Returns
+        -------
+        SplineXYZCurve: SplineXYZCurve
+            New representation of the curve parameterized by a spline for X,Y,Z.
+
+        """
+        from .curve import SplineXYZCurve
+
+        coords = self.compute("x", grid=grid, basis="xyz")["x"]
+        return SplineXYZCurve.from_values(
+            coords, knots=knots, method=method, name=name, basis="xyz"
+        )
+
+    # TODO: to_rz method for converting to FourierRZCurve representation
```

Comment:
We could probably add this, along with a check that phi/zeta is monotonic as a function of the original curve parameter, and throw a warning if its not?

Reply:
can be a new PR though

Reply:
Might save as another PR if I don't add in the next hour or so. 

I just realized, should I also be enforcing monotonicity of the curve s parameter? to ensure that the curve is always being traversed in one direction?

Reply:
it should be monotonic when the points are ordered "correctly", so maybe we just sort `s` and argsort `x` based on that?

Reply:
I think I am fine leaving it as is for now, and put the onus on the user to make sure it is monotonic, which most sensible use cases where user is supplying s, it should be